include/zephyr/devicetree/gpio.h

```diff
@@ -230,6 +230,180 @@ extern "C" {
 #define DT_GPIO_FLAGS(node_id, gpio_pha) \
 	DT_GPIO_FLAGS_BY_IDX(node_id, gpio_pha, 0)
 
+/**
+ * @brief Get the number of GPIO hogs in a node
+ *
+ * This expands to the number of hogged GPIOs, or zero if there are none.
+ *
+ * Example devicetree fragment:
+ *
+ *     gpio1: gpio@... {
+ *       compatible = "vnd,gpio";
+ *       #gpio-cells = <2>;
+ *
+ *       n1: node-1 {
+ *               gpio-hog;
+ *               gpios = <0 GPIO_ACTIVE_HIGH>, <1 GPIO_ACTIVE_LOW>;
+ *               output-high;
+ *       };
+ *
+ *       n2: node-2 {
+ *               gpio-hog;
+ *               gpios = <3 GPIO_ACTIVE_HIGH>;
+ *               output-low;
+ *       };
+ *     };
+ *
+ * Bindings fragment for the vnd,gpio compatible:
+ *
+ *     gpio-cells:
+ *       - pin
+ *       - flags
+ *
+ * Example usage:
+ *
+ *     DT_NUM_GPIO_HOGS(DT_NODELABEL(n1)) // 2
+ *     DT_NUM_GPIO_HOGS(DT_NODELABEL(n2)) // 1
+ *
+ * @param node_id node identifier; may or may not be a GPIO hog node.
+ * @return number of hogged GPIOs in the node
+ */
+#define DT_NUM_GPIO_HOGS(node_id) \
+	COND_CODE_1(IS_ENABLED(DT_CAT(node_id, _GPIO_HOGS_EXISTS)), \
+		    (DT_CAT(node_id, _GPIO_HOGS_NUM)), (0))
+/**
+ * @brief Get the node identifier for the controller phandle from a
+ *        GPIO hog node at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     gpio1: gpio@... {
+ *       compatible = "vnd,gpio";
+ *       #gpio-cells = <2>;
+ *
+ *       n1: node-1 {
+ *               gpio-hog;
+ *               gpios = <0 GPIO_ACTIVE_HIGH>, <1 GPIO_ACTIVE_LOW>;
+ *               output-high;
+ *       };
+ *
+ *       n2: node-2 {
+ *               gpio-hog;
+ *               gpios = <3 GPIO_ACTIVE_HIGH>;
+ *               output-low;
+ *       };
+ *     };
+ *
+ * Bindings fragment for the vnd,gpio compatible:
+ *
+ *     gpio-cells:
+ *       - pin
+ *       - flags
+ *
+ * Example usage:
+ *
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n1), 0) // DT_NODELABEL(gpio1)
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n1), 1) // DT_NODELABEL(gpio1)
+ *     DT_GPIO_HOG_CTLR_BY_IDX(DT_NODELABEL(n2), 0) // DT_NODELABEL(gpio1)
+ *
+ * @param node_id node identifier
+ * @param idx logical index into "gpios"
+ * @return the node identifier for the gpio controller referenced at
+ *         index "idx"
```

Comment:
The index has no bearing on the return value, so I'm a bit confused about why this macro needs the index as an argument. Doesn't `DT_PARENT(node_id)` suffice?

I'm also confused by why this macro isn't being used in gpio_hogs.c.

Reply:
You are right, `DT_PARENT(node_id)` would suffice (and I don't really need either due to the way I am iterating the dts nodes in `gpio_hogs.c`), but the `_PH` macros are generated as a side-effect of using `ControllerAndData` for storing the GPIO hogs in the DTS tooling.

Reply:
OK, if you want to expose it on the off-chance it will be useful, then I'd at least change the signature so it no longer requires the index.

Reply:
I have removed the `DT_GPIO_HOG_CTLR_BY_IDX` macro and added a note to `macros.bnf` that the generated `_PH` macros are currently unused by Zephyr.